utils/src/parallel_restore.c

```diff
@@ -1781,6 +1822,22 @@ spr_err_t scoutfs_parallel_restore_add_progress(struct scoutfs_parallel_restore_
 	return err;
 }
 
+spr_err_t scoutfs_parallel_restore_add_quota_rule(struct scoutfs_parallel_restore_writer *wri,
+						struct scoutfs_parallel_restore_quota_rule *rule)
+{
+	spr_err_t err;
+	if (!wri_has_super(wri)) {
+		err = EINVAL;
+		goto out;
+	}
+
+	err = insert_quota_item(wri, rule);
+out:
+	if (rule)
+		free(rule);
```

Comment:
No, I think this is bad API design. The library should stay consistent and not free pointers here. The data here could have been on the stack, or part of some other struct that the caller maintains, like an array.

The caller should free the `rule` after `scoutfs_parallel_restore_add_quota_rule()` returns. Ths same way that `scoutfs_parallel_restore_add_entry` and `scoutfs_parallel_restore_add_inode` also do not free the pointer.

In general, if the library allocates the memory, it should by default also free() it when it is appropriate. For stuff that's just passed in by pointer, it should by default not free the memory.